artichoke-backend/src/extn/core/array/backend/aggregate.rs

```diff
@@ -337,4 +339,18 @@ impl ArrayType for Aggregate {
         }
         Ok(Box::new(Self::with_parts(parts)))
     }
+
+    fn shuffle(&mut self, interp: &Artichoke) -> Result<(), Box<dyn RubyException>> {
+        let _ = interp;
+
+        let mut buffer = Vec::with_capacity(self.0.len());
+        for idx in 0..self.0.len() {
+            buffer.push(self.0.get(idx).unwrap());
+        }
+    
+        let mut rng = rand::thread_rng();
+        buffer.shuffle(&mut rng);
+        
+        Ok(())
```

Comment:
@lopopolo This doesn't work yet. The problem is my limited understanding of Rust... 
I don't understand how to convert the aggregate to a buffer and assign it to `self` (is that even what I should be doing?)...